### DISCUSSION TITLE
project home & install pages have serious defects

### DISCUSSION THREAD

**Discussion by shribe** [eclipse-birt/birt#1246] [Q&A]
As a potential new user, I started at https://eclipse.github.io/birt-website/

- The prominent "About", "Get started", and "Documentation" areas have nothing clickable. Not the graphics, not the titles, not the descriptions. No links anywhere.
- I'm in Safari, maybe that's the problem. Nope, same thing in Chrome and Edge.
- Check Javascript console, sure enough, a Javascript error down in jquery. In Chrome that is, try Safari, no error. Oh well, I'm not going to debug this.

Also: the prominent "Downloads" button takes me to a page which states "This page is not designed to be web-accessible" and is not useful to a new user. (More on this later.)

Anyway, after getting nowhere with the prominent, obviously intended to be entry points areas, I did notice the equivalent links at the top of the page. "Get started" takes me to https://eclipse.github.io/birt-website/docs/installation. Right off, it says "The BIRT download site is here" and gives a link to the same page as mentioned above. So let's try to explore that a bit:

- Link to directory "downloads" seems more like what I want than "update-site", it only has a "drops" directory, which then contains directories that are obviously versions, so I try the one for 4.13.0. Ah, we have zip'd packages for different architectures, but do I need rcp-report-designer or report-designer-all-in-one??? Well, anyway, back to the installation page to read the documentation about what packages to get and how to install them. Except of course, the installation page provides *NO SUCH INFORMATION*.
- Let's see, there's a link to the sources, which in its README has a link to the same downloads above. Anyway, source is probably not what I want at this stage. So let's continue reading. "Once you've downloaded your BIRT component, you need to install it. For instructions please continue reading..." Well sure, except I don't know what to download. So let's continue reading, maybe there's a hint coming.
- "If you have not yet done so, install Eclipse, GEF, EMF and BIRT. You'll find complete instructions for doing so on the BIRT Install page." Wait, what? I thought I was on the BIRT install page. And notably, there is no linked text here leading to a different install page.
- "Open BIRT. BIRT is a perspective within Eclipse...." Well, I actually know that, and how to open perspectives. The problem is, you kind of skipped over the whole part about how to find and install it!
- But the next section is titled "Designer Install", so maybe the installation instructions here are just not sequenced really well; let's keep going.
- "BIRT can be downloaded and installed in different ways:" Ahhh, this must be the section we've been looking for!
- First option, exactly what I think I'm looking for: "The complete BIRT designer or RCP Designer can be installed by downloading the required package here." *THERE IS NO LINK!* None of the text is linked, there is no link after that bullet point.
- Second option: "or by downloading the Eclipse IDE for Java and Report Developers Package." Not really what I want, since I already have an Eclipse installation with some packages for another project. But again, no link!
- Third option: "or through the update manager as described here..." Sure, I could do that,, but again, no link! I think I'll try using the update manager, but... What site to add???
- Then, hilariously, the docs proceed to hand-hold a user through creating a shortcut on the desktop to Eclipse.
  - Not all BIRT users will be Windows users.
  - Not all Windows users want to pollute their desktops with shortcuts to all the applications they use.
  - It seems unlikely that BIRT will attract novice users who don't know how launch their apps.
  - It is so incongruous to stick this in here, after failing in so many different ways to tell us how to install BIRT.
- Anyway, maybe "The Framework Designer Install" section will be better. But nope.
- "The Framework install requires that you already have the Eclipse Platform, GEF, EMF, DTP and WTP packages. The specific version of each, is listed on the specific build's download page." Really. What downloads page? No link, of course.
- "If you are installing the framework, you can download the BIRT framework and unzip the contents to any directory." Download what from where? Again, no link. Maybe I should just get that "report-designer-all-in-one", unzip it, and try???

After all that, I start wonder if maybe I did want "update-site" instead of "downloads" from that "yo, you shouldn't even be seeing this page" page. So back to that, "update-site" contains versioned folders. And **LO!** clicking on 4.13.0 gets me *install instructions*. At last. As far as I can tell, a standard page provided for Eclipse repos--nothing provided by BIRT. Except that when I add that repo to Eclipse, I seem to only have SDK packages, not the report developer.

So I have spent most of the morning on this, and still have almost no clue how to get started.

**Comment by speckyspooky** [comment]
To at first I say "Welocme to our community"!

Smal informations because you have asked many questions and the most are not caused on issues I would say it is normal instruction discussion to get started and that you will be familiar with BIRT and the aritecture.

The Web-Site is working and your clicks should be done on the head menu and it will work fine.
Yes, it xould be an improvment to change the titles which you reference to be cklickabel in future.

I woul prefer to use the "report-designer-all-in-one" because there you will get the full eclipse designer with the BIRT-installation at all.
The designer-version is the lightframe-version of the eclipse-framework.

Here you will get the current latest release but I think you figured out that:
https://download.eclipse.org/birt/downloads/drops/R-R1-4.13.0-202303022006/

I don't know what your final target is to integreate BIRT in an existing web application or use it standalone or so.
But with the all-in-one-designer you can start to create your first report.

**Comment by wimjongman** [comment]
Hey @shribe! Thanks for the write-up and also from me, welcome to the community. You are totally right, the instructions suck for newcomers. Like almost all open-source projects we are understaffed. I will fix the comments you made. We have a separate repository for the website at https://github.com/eclipse/birt-website please contribute if you have some spare time.
